back/app/services/idea_custom_fields_service.rb

```diff
@@ -198,5 +200,52 @@ def section1_title?(field, attribute)
     field.code == 'ideation_section1' && attribute == :title_multiloc
   end
 
+  def add_user_fields(fields)
+    return fields unless @participation_method.supports_user_fields_in_form?
+
+    fields = fields.to_a # sometimes array passed in, sometimes active record relations
+
+    last_page = fields.last
+    fields.pop # Remove the last page so we can add it back later
+
+    # Get the user fields from the permission (returns platform defaults if they don't exist)
+    phase = @custom_form.participation_context
+    permission = phase.permissions.find_by(action: 'posting_idea')
+    user_fields = Permissions::UserRequirementsService.new.requirements_custom_fields(permission)
+
+    # TODO: Hide any user fields that are locked for the user through the verification method
```

Comment:
Left because it's not really needed for the current use case and is difficult to do as we don't have the user context here